internal/validations/clusterlogging/validate_clusterlogging_spec.go

```diff
@@ -15,6 +21,16 @@ func validateClusterLoggingSpec(cl v1.ClusterLogging) error {
 		return nil
 	}
 	spec := cl.Spec
+	key := types.NamespacedName{Name: cl.Name, Namespace: cl.Namespace}
+	clf := &v1.ClusterLogForwarder{}
+	if err := k8sClient.Get(context.TODO(), key, clf); err != nil {
+		if apierrors.IsNotFound(err) {
+			return errors.NewValidationError("ClusterLogging: %s/%s instance requires to have a ClusterLogForwarder deployed in the same namespace named the same", cl.Namespace, cl.Name)
```

Comment:
We should consider a custom error message depending upon the usecase; and CLF is not always required

1. CL(openshift-logging/instance) only - This is a valid LEGACY usecase
1. CL(openshift-logging/instance) & CLF(openshift-logging/instance)   - This is a valid LEGACY usecase
1. CL(ANY_NS/OTHER) - This is an invalid usecase
1. CLF(ANY_NS/ANY_NAME) - This is a valid mCLF usecase
1. CL(ANY_NS/ANY_NAME) && CLF(ANY_NS/ANY_NAME) - This is a valid mCLF usecase

ANY_NAME is any name other then "instance" in "openshift-logging"

It looks like this still does not support the expected outcome for #1.  It may make more sense to refactor the method to try to load CLF first and then use a switch depending upon the outcome and ns/name combination:

```
clf, err := k8sClient.Get(NS, Name)
switch: {
case legacy CL only:
case legacy CL and CLF:
case mclf CL only:
case mclf CLF only:
case mCLF CLF and CL:
}
```